### ISSUE TITLE
rloadest won't fully install on mac

### ISSUE THREAD

**Issue by Kellyb90** [USGS-R/rloadest#37]
Hi there, I've been trying to install rloadest on my computer and I'm not having a lot of luck. I keep getting this error message related to smwrStats (pasted below). I have updated my R and Rstudio which didn't help. Any suggestions is greatly appreciated!

Warning message:
In i.p(...) :
  installation of package ‘/var/folders/dn/132b2sc97dv09dhw6_d4_4f00000gn/T//RtmpfyunLS/file8fc2770abeae/smwrStats_0.7.6.tar.gz’ had non-zero exit status

**Comment by ldecicco-USGS** [comment]
The first thing I would try is to choose "None" on any updates:
```r
remotes::install_github("USGS-R/smwrStats", force = TRUE)
Downloading GitHub repo USGS-R/smwrStats@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                                           
2: CRAN packages only                            
3: None                                          
4: pillar       (1.5.1      -> 1.6.0     ) [CRAN]
5: RcppArmad... (0.10.2.2.0 -> 0.10.4.0.0) [CRAN]
6: knitr        (1.31       -> 1.32      ) [CRAN]

Enter one or more numbers, or an empty line to skip updates: 3
√  checking for file 'C:\Users\ldecicco\AppData\Local\Temp\1\Rtmp8YO27G\remotese383e9a3ab9\USGS-R-smwrStats-2e02eb7/DESCRIPTION' (495ms)
-  preparing 'smwrStats': (1.3s)
√  checking DESCRIPTION meta-information ... 
```

If that doesn't work, could you paste in what you get when you run:
```r
devtools::session_info()
```

**Comment by Kellyb90** [comment]
Ok I tried choosing "none" on updates and that didn't work. This is what came up after running:
> devtools::session_info()
─ Session info ──────────────────────────────────────────────────────────────────────────────────────────────────
 setting  value                       
 version  R version 4.0.5 (2021-03-31)
 os       macOS High Sierra 10.13.6   
 system   x86_64, darwin17.0          
 ui       RStudio                     
 language (EN)                        
 collate  en_CA.UTF-8                 
 ctype    en_CA.UTF-8                 
 tz       America/Toronto             
 date     2021-04-14                  

─ Packages ──────────────────────────────────────────────────────────────────────────────────────────────────────
 package       * version    date       lib source                            
 abind           1.4-5      2016-07-21 [1] CRAN (R 4.0.2)                    
 akima           0.6-2.1    2020-05-30 [1] CRAN (R 4.0.2)                    
 cachem          1.0.4      2021-02-13 [1] CRAN (R 4.0.2)                    
 callr           3.6.0      2021-03-28 [1] CRAN (R 4.0.2)                    
 car             3.0-10     2020-09-29 [1] CRAN (R 4.0.2)                    
 carData         3.0-4      2020-05-22 [1] CRAN (R 4.0.2)                    
 cellranger      1.1.0      2016-07-27 [1] CRAN (R 4.0.2)                    
 cli             2.4.0      2021-04-05 [1] CRAN (R 4.0.2)                    
 colorspace      2.0-0      2020-11-11 [1] CRAN (R 4.0.2)                    
 crayon          1.4.1      2021-02-08 [1] CRAN (R 4.0.2)                    
 curl            4.3        2019-12-02 [1] CRAN (R 4.0.1)                    
 data.table      1.14.0     2021-02-21 [1] CRAN (R 4.0.2)                    
 dataRetrieval * 2.7.7      2021-03-06 [1] CRAN (R 4.0.2)                    
 desc            1.3.0      2021-03-05 [1] CRAN (R 4.0.2)                    
 devtools        2.4.0      2021-04-07 [1] CRAN (R 4.0.2)                    
 digest          0.6.27     2020-10-24 [1] CRAN (R 4.0.2)                    
 dplyr         * 1.0.5      2021-03-05 [1] CRAN (R 4.0.2)                    
 ellipsis        0.3.1      2020-05-15 [1] CRAN (R 4.0.2)                    
 fansi           0.4.2      2021-01-15 [1] CRAN (R 4.0.2)                    
 farver          2.1.0      2021-02-28 [1] CRAN (R 4.0.2)                    
 fastmap         1.1.0      2021-01-25 [1] CRAN (R 4.0.2)                    
 forcats         0.5.1      2021-01-27 [1] CRAN (R 4.0.2)                    
 forecast        8.14       2021-03-11 [1] CRAN (R 4.0.2)                    
 foreign         0.8-81     2020-12-22 [1] CRAN (R 4.0.5)                    
 fracdiff        1.5-1      2020-01-24 [1] CRAN (R 4.0.2)                    
 fs              1.5.0      2020-07-31 [1] CRAN (R 4.0.2)                    
 generics        0.1.0      2020-10-31 [1] CRAN (R 4.0.2)                    
 ggplot2       * 3.3.3      2020-12-30 [1] CRAN (R 4.0.2)                    
 ggtext          0.1.1      2020-12-17 [1] CRAN (R 4.0.2)                    
 glue            1.4.2      2020-08-27 [1] CRAN (R 4.0.2)                    
 gridtext        0.1.4      2020-12-10 [1] CRAN (R 4.0.2)                    
 gtable          0.3.0      2019-03-25 [1] CRAN (R 4.0.2)                    
 haven           2.3.1      2020-06-01 [1] CRAN (R 4.0.2)                    
 hms             1.0.0      2021-01-13 [1] CRAN (R 4.0.2)                    
 httr            1.4.2      2020-07-20 [1] CRAN (R 4.0.2)                    
 imputeTS        3.2        2021-01-16 [1] CRAN (R 4.0.2)                    
 jsonlite        1.7.2      2020-12-09 [1] CRAN (R 4.0.2)                    
 KernSmooth      2.23-18    2020-10-29 [1] CRAN (R 4.0.5)                    
 labeling        0.4.2      2020-10-20 [1] CRAN (R 4.0.2)                    
 lattice         0.20-41    2020-04-02 [1] CRAN (R 4.0.5)                    
 leaps           3.1        2020-01-16 [1] CRAN (R 4.0.2)                    
 lifecycle       1.0.0      2021-02-15 [1] CRAN (R 4.0.2)                    
 lmtest          0.9-38     2020-09-09 [1] CRAN (R 4.0.2)                    
 lubridate     * 1.7.10     2021-02-26 [1] CRAN (R 4.0.2)                    
 magrittr        2.0.1      2020-11-17 [1] CRAN (R 4.0.2)                    
 memoise         2.0.0      2021-01-26 [1] CRAN (R 4.0.2)                    
 munsell         0.5.0      2018-06-12 [1] CRAN (R 4.0.2)                    
 nlme            3.1-152    2021-02-04 [1] CRAN (R 4.0.5)                    
 nnet            7.3-15     2021-01-24 [1] CRAN (R 4.0.5)                    
 openxlsx        4.2.3      2020-10-27 [1] CRAN (R 4.0.2)                    
 pillar          1.6.0      2021-04-13 [1] CRAN (R 4.0.5)                    
 pkgbuild        1.2.0      2020-12-15 [1] CRAN (R 4.0.2)                    
 pkgconfig       2.0.3      2019-09-22 [1] CRAN (R 4.0.2)                    
 pkgload         1.2.1      2021-04-06 [1] CRAN (R 4.0.2)                    
 prettyunits     1.1.1      2020-01-24 [1] CRAN (R 4.0.2)                    
 processx        3.5.1      2021-04-04 [1] CRAN (R 4.0.2)                    
 ps              1.6.0      2021-02-28 [1] CRAN (R 4.0.2)                    
 purrr           0.3.4      2020-04-17 [1] CRAN (R 4.0.2)                    
 quadprog        1.5-8      2019-11-20 [1] CRAN (R 4.0.2)                    
 quantmod        0.4.18     2020-12-09 [1] CRAN (R 4.0.2)                    
 R6              2.5.0      2020-10-28 [1] CRAN (R 4.0.2)                    
 randtests       1.0        2014-11-17 [1] CRAN (R 4.0.2)                    
 Rcpp            1.0.6      2021-01-15 [1] CRAN (R 4.0.2)                    
 readxl          1.3.1      2019-03-13 [1] CRAN (R 4.0.2)                    
 remotes         2.3.0      2021-04-01 [1] CRAN (R 4.0.2)                    
 rio             0.5.26     2021-03-01 [1] CRAN (R 4.0.2)                    
 RiverLoad     * 1.0.2      2019-12-10 [1] CRAN (R 4.0.1)                    
 rlang           0.4.10     2020-12-30 [1] CRAN (R 4.0.2)                    
 rprojroot       2.0.2      2020-11-15 [1] CRAN (R 4.0.2)                    
 scales          1.1.1      2020-05-11 [1] CRAN (R 4.0.2)                    
 sessioninfo     1.1.1      2018-11-05 [1] CRAN (R 4.0.2)                    
 smwrBase      * 1.1.5      2021-04-13 [1] Github (USGS-R/smwrBase@2478663)  
 smwrData      * 1.1.2      2021-04-13 [1] Github (USGS-R/smwrData@c0a68c3)  
 smwrGraphs    * 1.1.4.9000 2021-04-13 [1] Github (USGS-R/smwrGraphs@f3debe7)
 sp              1.4-5      2021-01-10 [1] CRAN (R 4.0.2)                    
 stinepack       1.4        2018-07-30 [1] CRAN (R 4.0.2)                    
 stringi         1.5.3      2020-09-09 [1] CRAN (R 4.0.2)                    
 testthat        3.0.2      2021-02-14 [1] CRAN (R 4.0.2)                    
 tibble          3.1.0      2021-02-25 [1] CRAN (R 4.0.2)                    
 tidyr         * 1.1.3      2021-03-03 [1] CRAN (R 4.0.2)                    
 tidyselect      1.1.0      2020-05-11 [1] CRAN (R 4.0.2)                    
 timeDate        3043.102   2018-02-21 [1] CRAN (R 4.0.2)                    
 tseries         0.10-48    2020-12-04 [1] CRAN (R 4.0.2)                    
 TTR             0.24.2     2020-09-01 [1] CRAN (R 4.0.2)                    
 urca            1.3-0      2016-09-06 [1] CRAN (R 4.0.2)                    
 usethis         2.0.1      2021-02-10 [1] CRAN (R 4.0.2)                    
 utf8            1.2.1      2021-03-12 [1] CRAN (R 4.0.2)                    
 vctrs           0.3.7      2021-03-29 [1] CRAN (R 4.0.2)                    
 withr           2.4.1      2021-01-26 [1] CRAN (R 4.0.2)                    
 xml2            1.3.2      2020-04-23 [1] CRAN (R 4.0.2)                    
 xts             0.12.1     2020-09-09 [1] CRAN (R 4.0.2)                    
 zip             2.1.1      2020-08-27 [1] CRAN (R 4.0.2)                    
 zoo             1.8-9      2021-03-09 [1] CRAN (R 4.0.2)                    

[1] /Library/Frameworks/R.framework/Versions/4.0/Resources/library

**Comment by ldecicco-USGS** [comment]
Ah, yeah - I think there's some additional work that needs to happen for Macs. @wdwatkins or @aappling-usgs ...do either of you remember special installation instructions for Mac users?

**Comment by aappling-usgs** [comment]
I don't remember, sorry.

**Comment by wdwatkins** [comment]
`remotes::install_github("USGS-R/smwrStats", force = TRUE)` worked fine for me on my Mac (I skipped all package updates).  @Kellyb90  Can you post your install command and the full output?  I wonder if there is another error happening that would be more descriptive.

**Comment by Kellyb90** [comment]
@wdwatkins it didn't work but here is the full output:

> remotes::install_github("USGS-R/smwrStats", force = TRUE)
Downloading GitHub repo USGS-R/smwrStats@HEAD
These packages have more recent versions available.
It is recommended to update all of them.
Which would you like to update?

1: All                                           
2: CRAN packages only                            
3: None                                          
4: RcppArmad... (0.10.2.2.0 -> 0.10.4.0.0) [CRAN]

Enter one or more numbers, or an empty line to skip updates: 3
✓  checking for file ‘/private/var/folders/dn/132b2sc97dv09dhw6_d4_4f00000gn/T/Rtmpmhcxz6/remotes104163e425abd/USGS-R-smwrStats-2e02eb7/DESCRIPTION’ ...
─  preparing ‘smwrStats’:
✓  checking DESCRIPTION meta-information ...
─  cleaning src
─  checking for LF line-endings in source and make files and shell scripts (335ms)
─  checking for empty or unneeded directories
─  building ‘smwrStats_0.7.6.tar.gz’
   
* installing *source* package ‘smwrStats’ ...
** using staged installation
** libs
gfortran -mmacosx-version-min=10.13 -fno-optimize-sibling-calls  -fPIC  -Wall -g -O2  -c d_logam.f -o d_logam.o
make: gfortran: No such file or directory
make: *** [d_logam.o] Error 1
ERROR: compilation failed for package ‘smwrStats’
* removing ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/smwrStats’
* restoring previous ‘/Library/Frameworks/R.framework/Versions/4.0/Resources/library/smwrStats’
Warning message:
In i.p(...) :
  installation of package ‘/var/folders/dn/132b2sc97dv09dhw6_d4_4f00000gn/T//Rtmpmhcxz6/file1041676534e39/smwrStats_0.7.6.tar.gz’ had non-zero exit status

**Comment by wdwatkins** [comment]
> make: gfortran: No such file or directory

Looks like you need to install `gfortran` to be able to build the package, or specify the path to it if it's already installed.  I do recall messing with this some years ago and getting it to work, but not sure where I would have put the instructions...

Do you use homebrew, or otherwise know if you have gfortran?

**Comment by Kellyb90** [comment]
@wdwatkins I don't believe I have either. I am relatively new to R so I don't think I would of had a reason to install either of those. Can I download it for free somewhere?

**Comment by wdwatkins** [comment]
Yes, `gfortran` is an open-source fortran compiler, it generally comes with the `gcc` set of compilers.  Homebrew is a pretty popular package manager for Macs: https://brew.sh/.  There is a terminal command you can copy and paste there to install `brew`, which you can then use to install `gcc` with `brew install gcc` in the terminal.

**Comment by ldecicco-USGS** [comment]
I forgot I tried to set this up once. You all should try and let me know if it works:

```r
options(repos = c(
  ldeciccousgs = 'https://ldecicco-usgs.r-universe.dev',
  CRAN = 'https://cloud.r-project.org'))

# Install some packages
install.packages('rloadest')
```
I think it has a Mac binary which might solve these problems.

**Comment by IanGgit** [comment]
Hello, I'm having a very similar issue on a Mac and I wonder if there are any updates? I've tried everything suggested above.

I get:

Error: Failed to install 'smwrStats' from GitHub:
  Could not find tools necessary to compile a package
Call `pkgbuild::check_build_tools(debug = TRUE)` to diagnose the problem.

Then an RStudio message pops up:

Building R package from source requires installation of additional build tools.

Do you want to install the additional tools now?

Saying 'yes' opens this, which goes beyond my current capabilities:
https://www.cnet.com/tech/computing/install-command-line-developer-tools-in-os-x/

**Comment by sisik1967** [comment]
> I forgot I tried to set this up once. You all should try and let me know if it works:
> 
> ```r
> options(repos = c(
>   ldeciccousgs = 'https://ldecicco-usgs.r-universe.dev',
>   CRAN = 'https://cloud.r-project.org'))
> 
> # Install some packages
> install.packages('rloadest')
> ```
> 
> I think it has a Mac binary which might solve these problems.

Thanks. It worked on my windows pc.
